### PR TITLE
Main navbar container conversion to React

### LINKED ISSUE #2041: Convert navigation bar to react and keep all current functionality.
Convert navigation bar to react and keep all current functionality. 

Related to #2040

### PR DESCRIPTION
**Pull request by joykare**
Resolves #2041 

**Testing**
- Clone branch `joykare-main-navbar-2041`

**_Cart_**

- Click on the `cart button` on the navBar
- Expect checkout details to display on the top half of the page (empty cart  || products checkout)

**_Brand_**
_Test 1:_
- Navigate to the dashboard
- Select `Shop` and drag an image on `Drop file to upload`
- Observe the image displayed on the right of the brand name `REACTION`

_Test 2:_
- Navigate to random pages
- Click on Brand name 
- Expect a redirect to home screen

**_Sign out Dropdown_**

-  Click on different links and expect to have same experience as before

**_Whole  Navbar_**

- The whole nav-bar should work and look the same as before

### CODE REVIEW

**Comment by zenweasel** [comment]
It looks to me like the new and old versions are using some different font settings (the new version on the left looks larger and thicker?)
basic_reaction_product_and_basic_reaction_product_and_inbox__40__-_brent_thebuddhalodge_com_-_the_buddha_lodge_mail_

**Comment by zenweasel** [comment]
When I enable all currencies this bug appears to have reappeared
my_giant_but_otherwise_unremarkable_shop

**Comment by zenweasel** [comment]
It looks there is some discrepancies between how `development` and this branch display the nav bar in a mobile view. (`development` on right). I believe we are supposed to hide all icons except for search.
my_giant_but_otherwise_unremarkable_shop_and_my_giant_but_otherwise_unremarkable_shop_and_my_giant_but_otherwise_unremarkable_shop

**Comment by kieha** [comment]
@zenweasel my development navbar in mobile view looks like this: (the notification and search icons are the visible ones)

screen shot 2017-05-23 at 2 35 21 pm

**Comment by zenweasel** [comment]
@kieha Try changing your store name to be longer name and see if you get the same results I do

File: imports/plugins/included/default-theme/client/styles/navbar.less [hunk 1]
```diff
@@ -261,3 +247,57 @@ html:not(.rtl) .rui.navbar .languages .dropdown-menu {
...
+}
+
+.accounts-li-tag:last-child {
+  border-bottom: none;
+}
+
+.accounts-btn-tag {
+  padding: 6px 12px !important;
+  color: #fff !important;
+  background-color: @btn-primary-bg !important;
```
Comment (joykare) [inline imports/plugins/included/default-theme/client/styles/navbar.less]: To override the inherited styles.

**Comment by kieha** [comment]
@zenweasel you have to scroll the page to see the rest of the icons, but they're there. At least for me, they are.

screen shot 2017-05-23 at 3 50 30 pm

File: imports/plugins/included/default-theme/client/styles/navbar.less (hunk 1 above)
Comment (zenweasel) [inline imports/plugins/included/default-theme/client/styles/navbar.less]: Right, I understand what `important` does. Is there no way to make the selectors more specific? What happens when we want to override those styles?

**Pull request merged**